### PR TITLE
Upgrade to Ruby 3

### PR DESCRIPTION
**Pull request by indirect**
Okay, ready for review. This is... a lot. You might want to review it while pairing, and you might want to click the "commits" tab above, and review just one commit at a time.

- Update to Ruby 3.0 (from 2.6)
- Update to Rails 6.1 (from 5.2)
- Update many gems to support Ruby 3 & Rails 6.1
- Extract another controller out of a controller with two non-RESTful actions
- Make the country-choosing slightly less magic so that it is much simpler to test
- Completely stop using ajax on the "Get started" forms

The only _intentional_ change after this PR is that you now have to push "Next" after choosing a country, and that sends you to a page where you can create a plan for that country. If you want to change countries, hit the back button, change the country, hit next. You can't have ajax bugs if you don't use ajax! 🤔 

I'm hoping (but not sure) that this will be in a good enough state after review that someone can merge on Monday. On the other hand, if this too much to manage reviewing or fixing before I get back Thursday, go ahead and leave comments and I can work through them then.

### CODE REVIEW

**Comment by indirect** [comment]
huh. Semaphore 1.0 doesn't support Ruby 2.7 or 3.0, so this will have to wait until we merge the 2.0 changeover.

**Review by dysnomian — APPROVED** [review]

File: test/models/assessment_test.rb
Comment (indirect) [reply test/models/assessment_test.rb:9]: This change resolves a deprecation warning printed by the latest version of Minitest.

Minitest decided that to stop monkey patching all objects with a `must_equal` method, they would instead define a method named `_` that returns a minitest assertion wrapper around the thing you pass to the underscore. The minutest assertion wrapper object is where the `must_equal` method lives.

**Pull request merged**
